### ISSUE TITLE
Lint/ShadowingOuterLocalVariable: a false positive case with if/eslif/else statement and block variable

### ISSUE THREAD

**Issue by stefanoc** [rubocop/rubocop#10962]
## Expected behavior

No offense with Lint/ShadowingOuterLocalVariable cop 

## Actual behavior

```
 W: Lint/ShadowingOuterLocalVariable: Shadowing outer local variable - item.
    items.each { |item| frobnicate(item) }
                  ^^^^
```

## Steps to reproduce the problem

```
def do_something(items, what)
  if what == :first
    item = items[0]
    frobnicate(item)
  elsif what == :second
    item = items[1]
    frobnicate(item)
  else
    items.each { |item| frobnicate(item) }
  end
end

rubocop --only Lint/ShadowingOuterLocalVariable
```

Removing the `elsif` does not cause the warning to be emitted (see #10479)

## RuboCop version

```
%% rubocop -V
1.35.1 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.1.2 x86_64-darwin21)
```
